proxy/src/coco.rs

```diff
@@ -55,9 +57,30 @@ impl Peer {
         I: Iterator<Item = (peer::PeerId, SocketAddr)> + Send + 'static,
     {
         let peer = config.try_into_peer().await?;
-        // TODO(finto): discarding the run loop below. Should be used to subsrcibe to events and
-        // publish events.
-        let (api, _futures) = peer.accept()?;
+
+        let (api, run_loop) = peer.accept()?;
+
+        let protocol = api.protocol();
+        let protocol_subscriber = protocol.subscribe().await;
+        let protocol_notifications = protocol_subscriber.for_each(|notification| {
+            trace!("PROTOCOL: {:?}", notification);
```

Comment:
If we leave these in, maybe we can be more helpful, eg. `trace!("Received protocol notification: {:?}, notif)`